apps/api/v2/src/modules/workflows/inputs/create-workflow.input.ts

```diff
@@ -0,0 +1,229 @@
+  @ApiProperty({ description: "Time value for offset before/after event trigger", example: 24, type: Number })
+  @IsNumber()
+  value!: number;
+
+  @ApiProperty({ description: "Unit for the offset time", example: "HOUR", enum: WorkflowTimeUnit })
+  @IsEnum(WorkflowTimeUnit)
+  unit!: WorkflowTimeUnit;
+}
+
+export class WorkflowTriggerDto {
+  @ApiProperty({
+    description: "Trigger type for the workflow",
+    example: "BEFORE_EVENT",
+    enum: WorkflowTriggerType,
+  })
+  @IsEnum(WorkflowTriggerType)
+  type!: WorkflowTriggerType;
+
+  @ApiPropertyOptional({
+    description: "Offset before/after the trigger time; required for BEFORE_EVENT and AFTER_EVENT only",
+    type: WorkflowTriggerOffsetDto,
+    required: false,
+  })
+  @ValidateIf((o) =>
+    Boolean(
+      o.type && (o.type === WorkflowTriggerType.BEFORE_EVENT || o.type === WorkflowTriggerType.AFTER_EVENT)
+    )
+  )
+  @ValidateNested()
+  @Type(() => WorkflowTriggerOffsetDto)
+  @IsOptional()
+  offset?: WorkflowTriggerOffsetDto;
+}
+
+export class WorkflowMessageDto {
+  @ApiProperty({
+    description: "Subject of the message",
+    example: "Reminder: Your Meeting {EVENT_NAME} - {EVENT_DATE_ddd, MMM D, YYYY h:mma} with Cal.com",
+  })
+  @IsString()
+  subject!: string;
+
+  @ApiProperty({
+    description: "HTML content of the message",
+    example:
+      "<p>This is a reminder from {ORGANIZER} of {EVENT_NAME} to {ATTENDEE} starting here  {LOCATION} {MEETING_URL} at {START_TIME_h:mma} {TIMEZONE}.</p>",
+    required: false,
+  })
+  @IsOptional()
+  @IsString()
+  html?: string;
+
+  @ApiProperty({
+    description: "Text content of the message (used for SMS)",
+    example:
+      "This is a reminder message from {ORGANIZER} of {EVENT_NAME} to {ATTENDEE} starting here {LOCATION} {MEETING_URL} at {START_TIME_h:mma} {TIMEZONE}.",
+    required: false,
+  })
+  @IsOptional()
+  @IsString()
+  text?: string;
+}
+
+export class WorkflowStepDto {
+  @ApiProperty({ description: "Unique identifier of the step", example: 67244 })
+  @IsNumber()
+  id!: number;
+
+  @ApiProperty({ description: "Step number in the workflow sequence", example: 1 })
+  @IsNumber()
+  stepNumber!: number;
+
+  @ApiProperty({ description: "Action to perform", example: "EMAIL_HOST", enum: StepAction })
+  @IsEnum(StepAction)
+  action!: StepAction;
+
+  @ApiProperty({ description: "Recipient type", example: "ATTENDEE", enum: RecipientType })
+  @IsEnum(RecipientType)
+  recipient!: RecipientType;
+
+  @ApiPropertyOptional({
+    description: "Email address if recipient is EMAIL, required for action EMAIL_ADDRESS",
+    example: "31214",
+    required: false,
+    externalDocs: {
+      url: "https://cal.com/docs/api-reference/v2/organization-team-verified-resources/verify-an-email-for-an-org-team",
+    },
+  })
+  @ValidateIf((o) => Boolean(o.action && o.action === StepAction.EMAIL_ADDRESS))
+  @IsNumber()
+  verifiedEmailId?: number;
+
+  @ApiPropertyOptional({
```

Comment:
Here too i think we need separate dtos for email and sms - makes it easier in docs to see complete object of whats expected rather than checking if property is required or not.